### PR TITLE
Docs fixes listify

### PR DESCRIPTION
**Pull request by willkg**
* implement listify and minor code cleanup
* add short-circuit to override environment
* docs updates and cleanup
* add note about django-cache-url

### CODE REVIEW

**Comment by willkg** [comment]
@pmac Does this look to you?

File: HISTORY.rst [hunk 1]
```diff
@@ -4,13 +4,19 @@ History
...
 
-* (Bug) Fix ``__call__`` returning None--should return ``NO_VALUE``
-* Make ``NO_VALUE`` falsy
-* Finish docs about making your own parsers
-* Add section to docs about using Everett with dj-database-url
+* Feature: Add ``ConfigEnvFileEnv`` for supporting ``.env`` files. (Thank you,
+  Paul!)
+* Feature: Change ``ConfigIniEnv`` to take a single path or list of paths.
+  (Thank you, Paul!)
+* Feature: Make ``NO_VALUE`` falsy. (Thank you, Paul!)
```
Comment (pmac) [inline HISTORY.rst]: You did that though :)

File: docs/configuration.rst [hunk 2]
```diff
@@ -77,18 +75,26 @@ You can implement your own sources. They just need to implement the ``.get()``
...
         def get(self, key, namespace=None):
-            # The namespace is either None or a list of strings specifying
-            # the namespace.
-            if namespace is None:
-                namespace = []
+            # The namespace is either None, a string or a list of strings.
+            if namespace:
+                # This forces the namespace value to always be a list of
+                # strings
+                namespace = listify(namespace)
```
Comment (pmac) [inline docs/configuration.rst]: Why do this at all in the `NoOpEnv`?

Comment (pmac) [inline docs/configuration.rst]: Ah. Just as example. Carry on.

File: docs/configuration.rst [hunk 3]
```diff
@@ -266,6 +272,37 @@ That'll pull the ``DATABASE_URL`` value from the environment (it throws an error
...
+Everett works great with `django-cache-url <https://github.com/ghickman/django-cache-url>`_.
+
+For example::
+
+    import django_cache_url
+    from everett.manager import ConfigManager, ConfigOSEnv
+
+    config = ConfigManager([ConfigOSEnv()])
+    CACHES = {
+        'default': config('CACHE_URL', parser=django_cache_url.parse)
```
Comment (pmac) [inline docs/configuration.rst]: I think I'd throw a `default` on either this one or the `dj_database_url` one just to demonstrate a slightly more interesting default. For the database one e.g. you could use `config('DATABASE_URL', parser=dj_database_url.parse, default='sqlite:///my.db')`.

File: everett/manager.py [hunk 4]
```diff
@@ -263,17 +292,30 @@ def get(self, key, namespace=None):
...
+
+        config = ConfigManager([
+            ConfigIniEnv(os.environ.get('FOO_INI'))
+        ])
+
+
+    If there's no ``FOO_INI`` in the environment, then the path will be
+    ignored.
+
+    Here's an example that looks for the INI filespecified in the environment
```
Comment (pmac) [inline everett/manager.py]: typo: `filespecified`

**Comment by pmac** [comment]
Just a couple of nits to update if you'd like. Looks great on the whole. 👍

File: docs/configuration.rst (hunk 3 above)
Comment (willkg) [inline docs/configuration.rst]: That's a good idea. I'll add that.

**Comment by willkg** [comment]
^^^ That I think covers everything. Travis is happy, so I'm going to merge it.

Thank you! This was a big help!

**Pull request merged**